### ISSUE TITLE
Feature computation for large 1d-masks produces 'nan' feature values

### ISSUE THREAD

**Issue by maxschloegel** [DC-analysis/dcnum#9]
In some cases, the mask of segmented events is 1-dimensional. This case is not skipped for feature computation which leads to NaN-values in the features of the output RT-DC file.

**Comment by maxschloegel** [comment]
Currently in `dcnum.feat.feat_moments.mt_legacy.py` the function `moments_based_features` for feature computation checks for small 1-dimensional events but large events are not yet dealt with robustly. Therefore my suggested solution is to add a test for the area of the event via
```python
import cv2
contour = ...
area = cv2.contourArea(countour)

# skip event if area is zero
if area == 0:
    continue
```
This should deal remove the currently problematic cases.

**Comment by maxschloegel** [comment]
After implementing, I realized that the `NaN`-entries that are resulting from `continue`  in the `for`-loop might not be properly removed, but are kept in the events.

What would be the best way to remove the `NaN`-entries of the `legacy`-features?
Could we use the gate-class to do that or should we maybe pass a `keep_event` boolean array (which is set to False for each event with nan-values) from the function `moments_based_features` and then remove all events where this array is `False`?
```python
.
.
.
    inert_ratio_raw = np.copy(empty)
    inert_ratio_prnc = np.copy(empty)
    keep_event = np.full(size, True)

    for ii in range(size):
        cont_raw = contour_single_opencv(mask[ii])
        if len(cont_raw.shape) < 2:
            keep_event[ii] = False
            continue
        if cv2.contourArea(cont_raw) == 0:
            keep_event[ii] = False
            continue
        mu_raw = cv2.moments(cont_raw)

        # convex hull
        cont_cvx = np.squeeze(cv2.convexHull(cont_raw))
        mu_cvx = cv2.moments(cont_cvx)

        if mu_cvx["m00"] == 0 or mu_raw["m00"] == 0:
            # Contour size too small
            keep_event[ii] = False
            continue
```

**Comment by maxschloegel** [comment]
Or maybe we can add a default-Gate that is always checking one of the legacy-features for a nan-value and automatically discards all `nan`-events?

**Comment by paulmueller** [comment]
Sounds like a good plan. Instead of "not_nan", I would use "invalid" which does not have twice the negation and it also describes the purpose of this step.

Also, one could do an `if np.any(gated_events["invalid"]` before applying the for-loop.

**Comment by paulmueller** [comment]
In case you would like to work on a PR, make sure to increment the `DCNUM_PPID_GENERATION` in `dcnum.meta.ppid`, since this is a breaking change for pipeline reproducibility.

**Comment by maxschloegel** [comment]
Ok thanks! I changed the code accordingly.

**Comment by paulmueller** [comment]
Fixed in 0.12.1
